desktop/cmp/grouping/GroupingChooser.ts

```diff
@@ -292,6 +284,28 @@ function getDimOptions(dims, model) {
     return sortBy(ret, 'label');
 }
 
+function targetIsControlButtonOrPortal(isOpen, e, model, nextOpenState) {
+    if (isOpen && nextOpenState === false) {
+        // Prevent clicks with Select controls from closing popover
+        const id = MENU_PORTAL_ID,
+            selectPortal = document.getElementById(id)?.contains(e?.target),
+            selectClick = e?.target?.classList.contains('xh-select__single-value');
+        // Determines if grouping-chooser is a parent - if not, close popover
+        let editorClick = false;
+        let elem = e?.target;
+        while (elem) {
```

Comment:
think if you had a method `targetWithin(className)` that would really help you here.
It could also be easily be implemented with a simple loop, walking up the parent chain until the root, and returning early if the class in question was found.

Reply:
see our `isDisplayed()` method in DomUtils for an example of something similar